Lib/test/test_posix.py

```diff
@@ -2172,6 +2172,58 @@ def test_utime(self):
                 os.utime("path", dir_fd=0)
 
 
+class NamespacesTests(unittest.TestCase):
+    """Tests for os.unshare() and os.setns()."""
+
+    @support.requires_subprocess()
+    def subprocess(self, code):
+        import subprocess
+        with subprocess.Popen((sys.executable, '-c', code),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            encoding="utf-8"
+        ) as p:
+            p.wait()
+            return (
+                p.returncode,
+                tuple(p.stdout),
+                tuple(p.stderr),
+            )
+
+    @unittest.skipUnless(hasattr(os, 'unshare'), 'needs os.unshare()')
+    @unittest.skipUnless(hasattr(os, 'setns'), 'needs os.setns()')
+    @unittest.skipUnless(os.path.exists('/proc/self/ns/uts'), 'need /proc/self/ns/uts')
+    @support.requires_linux_version(3, 0, 0)
+    def test_unshare_setns(self):
+        rc, out, err = self.subprocess("""if 1:
```

Comment:
Heh, nice hack. There is actually an easier way: https://docs.python.org/3/library/textwrap.html#textwrap.dedent

It might be easier to turn the test case into an external script. It would make debugging issues easier, too.

Reply:
Keep in mind that the `if 1:` hack avoids an import.

Reply:
IMO, I think it is a bad idea to move this out to a separate file. There are hundreds of subprocess tests in the test suite. We do not want to pollute Lib/test with a lot of subprocess scripts. At least create a posix subdirectory and move it there.

Reply:
@erlend-aasland which one do you like better? external script in a dedicated subdirectory, or an inline script (with `if 1:` or `textwrap.dedent`)? seems like most of the tests are using an inline script.

Reply:
I definitely prefer inline. And if you inline it, it is my personal opinion that we prioritise saving an import by using the poor-mans-dedent. Keeping the number of imports in the test suite low is a good thing, IMO.

However, I won't go into a larger debate about this. I've delivered my opinions, and I'll refrain from commenting further, unless absolutely needed :)